### PR TITLE
use bcrypt, add a pyshop-user response header.

### PR DESCRIPTION
**Pull request by jaredg**
A while ago, cryptacular got updated in a way that I couldn't get the C bits to build in our environment so i ported it to pure-python bcrypt.  

This also adds a custom HTTP response header that I can scrape with uWSGI to see who is actually using our pyshop.

### CODE REVIEW

**Comment by Roguelazer** [comment]
you tested to make sure that the encoding formats are the same so the new bcrypt library can validate passwords encoded using the old one?

**Comment by jaredg** [comment]
> you tested to make sure that the encoding formats are the same so the new bcrypt library can validate passwords encoded using the old one?

Yes, it reads the old salt parameters okay.  I tried both downloads and uploading a new version of a package.

File: pyshop/models.py [hunk 1]
```diff
@@ -167,8 +166,13 @@ def name(self):
...
         return self._password
 
+    @classmethod
+    def hash_password(cls, pw):
+        return bcrypt.hashpw(pw.encode('utf8'), bcrypt.gensalt(12))
+
     def _set_password(self, password):
-        self._password = unicode(crypt.encode(password))
+        # we want to store this in the database as unicode
+        self._password = unicode(hashpw(password))
```
Comment (Roguelazer) [inline pyshop/models.py:175]: nit: using `unicode(obj)` to go from bytes to unicode is unsafe (and results in calling `repr` if you do the equivalent in python 3000) and relies on `$LOCALE` (so could do weird things in weird locals); prefer `obj.decode('ascii')` if that is what you want

**Review by Roguelazer — APPROVED** [review]

File: pyshop/models.py (hunk 1 above)
Comment (jaredg) [reply pyshop/models.py:175]: https://github.com/EasyPost/pyshop/blob/master/pyshop/compat.py#L6

I think this existing monkeypatch is probably even worse but it's used in a lot of places.  If I decide to move forward with py3 I'll just take this out.

**Pull request merged**